sympy/solvers/decompogen.py

```diff
@@ -42,7 +43,7 @@ def decompogen(f, symbol):
 
     # ===== Convert to Polynomial ===== #
     fp = Poly(f)
-    gens = fp.gens
+    gens = list(filter(lambda x: symbol in x.atoms(Symbol) , fp.gens))
```

Comment:
Oh! `filter` will return tuple only in Python 2 otherwise it will return an iterator. +1